hazelcast/partition.py

```diff
@@ -55,7 +57,12 @@ def get_partition_id(self, key: typing.Any) -> int:
         Returns:
             The partition id.
         """
-        key_data = self._serialization_service.to_data(key)
+        try:
+            key_data = self._serialization_service.to_data(key)
+        except SchemaNotReplicatedError as e:
+            self._send_schema_and_retry_fn(e, lambda: ...).result()
+            return self.get_partition_id(key)
```

Comment:
What do you think about doing what I did in Node.js? I mean we don't need to send the schema to the cluster for this operation because the data is not sent to the cluster. It will affect the performance of this method but don't know how much important it is.

Reply:
I think this method is not important, and I don't want to introduce a new parameter to to_data